services/gitea/gitea-languages-count.service.js

```diff
@@ -0,0 +1,79 @@
+import Joi from 'joi'
+import { nonNegativeInteger, optionalUrl } from '../validators.js'
+import { metric } from '../text-formatters.js'
+import { pathParam, queryParam } from '../index.js'
+import { documentation, httpErrorsFor } from './gitea-helper.js'
+import GiteaBase from './gitea-base.js'
+
+/*
+We're expecting a response like { "Python": 39624, "Shell": 104 }
+The keys could be anything and {} is a valid response (e.g: for an empty repo)
+*/
+const schema = Joi.object().pattern(/./, nonNegativeInteger)
+
+const queryParamSchema = Joi.object({
+  gitea_url: optionalUrl,
+}).required()
+
+export default class GiteaLanguageCount extends GiteaBase {
+  static category = 'analysis'
+
+  static route = {
+    base: 'gitea/languages/count',
+    pattern: ':user/:repo',
+    queryParamSchema,
+  }
+
+  static openApi = {
+    '/gitea/languages/count/{user}/{repo}': {
+      get: {
+        summary: 'Gitea language count',
+        description: documentation,
+        parameters: [
+          pathParam({
+            name: 'user',
+            example: 'go-gitea',
+          }),
+          pathParam({
+            name: 'repo',
+            example: 'gitea',
+          }),
+          queryParam({
+            name: 'gitea_url',
+            example: 'https://codeberg.org',
+          }),
+        ],
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'languages' }
+
+  static render({ languagesCount }) {
+    return {
+      message: metric(languagesCount),
+      color: 'blue',
+    }
+  }
+
+  async fetch({ user, repo, baseUrl }) {
+    // https://codeberg.org/api/swagger#/repository/repoGetLanguages
+    return super.fetch({
+      schema,
+      url: `${baseUrl}/api/v1/repos/${user}/${repo}/languages`,
+      httpErrors: httpErrorsFor('user or repo not found'),
+    })
+  }
+
+  async handle(
+    { user, repo },
+    { gitea_url: baseUrl = 'https://codeberg.org' },
```

Comment:
Defaulting GitLab badges to gitlab.com makes sense because it is the "official" GitLab instance
Is codeberg the "canonical" implementation of gitea, or is it just the most popular public install?
Feels like the default for this should either be GiTea Cloud, or if it is kind of ambiguous, we should just make `gitea_url` a required query param with no default.

Reply:
As discussed, `gitea_url` will be required as there is no canonical implementation, but tests can use codeberg.com still.